velox/exec/tests/MarkDistinctTest.cpp

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/exec/tests/utils/AssertQueryBuilder.h"
+#include "velox/exec/tests/utils/OperatorTestBase.h"
+#include "velox/exec/tests/utils/PlanBuilder.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::test;
+using namespace facebook::velox::exec::test;
+
+class MarkDistinctTest : public OperatorTestBase {
+ public:
+  void runBasicMarkDistinctTest(const VectorPtr& base) {
+    std::vector<RowVectorPtr> vectors;
+
+    const vector_size_t baseSize = base->size();
+    const vector_size_t size = baseSize * 2;
+    auto indices = makeIndices(size, [&](auto row) { return row % baseSize; });
+    auto baseEncoded = wrapInDictionary(indices, size, base);
+
+    vectors.push_back(makeRowVector({baseEncoded}));
+
+    auto distinctCol =
+        makeFlatVector<bool>(size, [&](auto row) { return row < baseSize; });
+
+    auto expectedResults = makeRowVector({baseEncoded, distinctCol});
+
+    auto op = PlanBuilder()
+                  .values(vectors)
+                  .markDistinct("c0$Distinct", {"c0"})
+                  .planNode();
+
+    auto results = AssertQueryBuilder(op).copyResults(pool());
+    assertEqualVectors(results, expectedResults);
+  }
+};
+
+template <typename T>
+class MarkDistinctPODTest : public MarkDistinctTest {};
+
+using MyTypes =
+    ::testing::Types<int8_t, int16_t, int32_t, int64_t, float, double, bool>;
+TYPED_TEST_SUITE(MarkDistinctPODTest, MyTypes);
+
+TYPED_TEST(MarkDistinctPODTest, basic) {
+  auto data = VectorTestBase::makeFlatVector<TypeParam>(
+      2, [&](auto row) { return row % 2; });
```

Comment:
I noticed that these tests use very small dataset with just 2 rows. I changed this to test with 1000 rows.